### PR TITLE
Split iapp_template into two plugins

### PR DESCRIPTION
**Pull request by pjbreaux**
@zancas 
#### What's this change do?

The f5_sys_iapptemplate plugin needs to be split into two plugins: one for a full template upload, and the other for a composite template.

There's a great deal of changes in the yaml templates. Mostly copy paste and changing file names. The other fixups are to migrate from one plugin to two.
#### Any background context?

This is to be done for usability and readability, as they are two different starting points that reach the same end (creating an iapp template on the bigip). When these two plugins were squashed into one plugin, there was extra logic to facilitate both types of input. This was becoming cumbersome and difficult to read.

@swormke: Wondering about how docs should be handled for this new plugin?
#### Where should the reviewer start?

Start at the unit tests

### CODE REVIEW

**Comment by pjbreaux** [comment]
Bumping this to @zancas now that shawn has fixed the slack integration

File: f5_heat/resources/f5_ltm_pool.py [hunk 1]
```diff
@@ -86,6 +86,7 @@ class F5LTMPool(resource.Resource, F5BigIPMixin):
         )
     }
 
+    @f5_common_resources
```
Comment (zancas) [inline f5_heat/resources/f5_ltm_pool.py:89]: Consider moving the definition of '`f5_common_resources`' into '`F5BigIPMixin`', and calling it as an instance method inside the method it is currently decorating, e.g.:
`self._f5_common_resources`.

Also, its name should probably start with '_'.

Comment (zancas) [inline f5_heat/resources/f5_ltm_pool.py:89]: One advantage of moving `f5_common_resources` into a mixin is that it eliminates an import statement.
Also it should be renamed to start with `_VERB`.

File: f5_heat/resources/f5_sys_iappcompositetemplate.py [hunk 2]
```diff
@@ -125,13 +92,13 @@ def _add_optional_attr(self, iapp_dict):
         :returns: possibly modified dictionary
         '''
 
-        if self.properties[self.COMPOSITE_TEMPLATE][self.REQUIRES_MODULES]:
+        if self.properties[self.REQUIRES_MODULES]:
```
Comment (zancas) [inline f5_heat/resources/f5_sys_iappcompositetemplate.py]: I don't think this pattern will do what you intend.   I think you will get what you intend if you use the pattern:

`if self.REQUIRES_MODULES in self.properties:`   etc.

File: f5_heat/resources/f5_sys_iappcompositetemplate.py [hunk 3]
```diff
@@ -125,13 +92,13 @@ def _add_optional_attr(self, iapp_dict):
...
             iapp_dict['requiresModules'] = \
```
Comment (zancas) [inline f5_heat/resources/f5_sys_iappcompositetemplate.py:96]: Is it possible to specify `self.properties` such that this conditional can be replaced with:  `iapp_dict.update(self.properties)` ?

File: f5_heat/resources/f5_sys_iappcompositetemplate.py [hunk 4]
```diff
@@ -142,10 +109,8 @@ def _build_iapp_dict(self):
         '''
 
         sections = {
-            'implementation': self.properties[
-                self.COMPOSITE_TEMPLATE][self.IMPLEMENTATION] or '',
-            'presentation': self.properties[
-                self.COMPOSITE_TEMPLATE][self.PRESENTATION] or ''
+            'implementation': self.properties[self.IMPLEMENTATION] or '',
+            'presentation': self.properties[self.PRESENTATION] or ''
```
Comment (zancas) [inline f5_heat/resources/f5_sys_iappcompositetemplate.py:113]: Are the referenced properties "str"s ?

Comment (pjbreaux) [inline f5_heat/resources/f5_sys_iappcompositetemplate.py:113]: Yes.

**Pull request merged**